fastembed/text/onnx_embedding.py

```diff
@@ -217,11 +227,29 @@ def embed(
             batch_size=batch_size,
             parallel=parallel,
         )
+    def onnx_embed(self, documents: List[str]) -> Tuple[np.ndarray, np.ndarray]:
+        encoded = self.tokenizer.encode_batch(documents)
+        input_ids = np.array([e.ids for e in encoded])
+        attention_mask = np.array([e.attention_mask for e in encoded])
+
+        # Prepare the ONNX input dictionary
+        onnx_input = {
+            "input_ids": input_ids.astype(np.int64),
+            "attention_mask": attention_mask.astype(np.int64),
+        }
 
+        # Check if 'token_type_ids' is expected by the model
+        if 'token_type_ids' in [input.name for input in self.model.get_inputs()]:
+            token_type_ids = np.zeros_like(input_ids)
+            onnx_input["token_type_ids"] = token_type_ids.astype(np.int64)
+
+        model_output = self.model.run(None, onnx_input)
+        embeddings = model_output[0]
+        return embeddings, attention_mask
```

Comment:
We can override `_preprocess_onnx_input()` instead.

For example https://github.com/qdrant/fastembed/blob/main/fastembed%2Ftext%2Fe5_onnx_embedding.py#L45-L50

Reply:
@Anush008 ,Initially I did that only but the error  persists then only  I modified [onnx_embedding.py ](https://github.com/qdrant/fastembed/pull/197/files#diff-c3531be71315167c13fae0ab3afeb93d52ec56bf9f004e4a60675e369530c4e8)

Reply:
You could also try it locally

Reply:
@Anush008 ,It won't work as `e5_onnx_embedding.py` is  different. The model is being stored in `onnx_embedding.py`. It will work for an e5 model. The same error occurred while uploading  `e5-small` and it resolved  by modifying the `  def _preprocess_onnx_input` . But for `bge-m3` it isn't working.

Reply:
for `multilingual-e5-small model` , I override like this and it worked :  

    def _preprocess_onnx_input(self, onnx_input: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
        """
        Preprocess the onnx input.
        """
        if "token_type_ids" not in self.required_inputs:
            onnx_input.pop("token_type_ids", None)
        return onnx_input

But it failed for `bge-m3`

Reply:
https://github.com/qdrant/fastembed/pull/190/files#diff-d029a94e189b08dcd60a4efc05e94ce6af8f51ebbda58eecdc6413e0d439eb14

Reply:
[CI logs](https://github.com/qdrant/fastembed/actions/runs/8626874505/job/23645811364#step:5:21)

Reply:
You're saying overriding `_preprocess_onnx_input()` in `onnx_embedding.py` doesn't work?

Reply:
I tried it another way ,now it's working .All tests have passed